site/content/posts/2021-08-12-octant-0.23-release.md

```diff
@@ -0,0 +1,91 @@
+
+While there are five possible values for Pod phase:
+
+| Value | Description |
+| :----:  | ----- |
+| Pending | Configuration accepted by the server but not ready to run (e.g. init containers, pending image fetching, among others) |
+| Running | All of the Pod's containers created, at least one of them is running|
+| Succeeded | All containers successfully completed their task and then exited, especially true in the presence of OnFailure [restartPolicy](https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle/#restart-policy) |
+| Failed | All containers terminated and at least one terminated in failure|
+| Unknown| Most likely the node is offline |
+
+There are also three container states defined in the Kubernetes API:
+
+| Value| Description|
+| ----| ----|
+|Waiting|   A task is pending to complete, for example a container image download or an init container still running|
+|Running| Container executing with no issues nor pending tasks|
+|Terminated| Container either ran to completion or it failed
+
+So, coming back to our example: what kubectl is showing for the "activator" Pod is the **Phase** field (in this case **Running** as the container has been created) but the application there it's not usable yet. How could we determine more accurately the actual situation of the Pod?
+
+Let's see:
+
+# Octant and the Status field
+
+Octant has always displayed the **Phase** field for each resource, but with the release of version 0.23, the team added the **Status** field which -in the case of Pods- refers specifically to the container State. 
+
+So, if we take a look at the "activator" Pod with Octant:
+
+![](/img/posts/2021/08/18/octantStatus01.png)
+
+We can see that it's in the **Running** phase but the actual container state is **CrashLoopBackOff** which means that the Kubelet has succesfully started it several times but it has crashed afterwards without reaching a stable state yet.
+
+We can get quickly to the root cause of the issue with Octant:
+
+1. Click on the Pod name
+2. Go to the **Resource Viewer** tab and there you will find the **Last Event** field:
+
+![](/img/posts/2021/08/18/octantStatus02.png)
+
+3. Finally, go to the **Logs** tab to gather even more details around the Readiness Probe failure:
+```json
+    {"severity":"ERROR","timestamp":"2021-08-31T21:08:56.7648856Z","logger":"activator","caller":"websocket/connection.go:145","message":"Websocket connection could not be established","commit":"c75484e","knative.dev/controller":"activator","knative.dev/pod":"activator-67656dcbbb-hq62r","error":"dial tcp 10.96.38.80:8080: connect: connection refused"}
+```
+There it is what seems to be root cause of the issue: a network connectivity problem. 
```

Comment:
That's not a root cause. Why is the connection refused?

Reply:
Still working on this